### PR TITLE
Af 284 pl liquidation flow initiated

### PR DESCRIPTION
**Pull request by acollette**
This PR accomplishes the following:

-Remove `liquidateAccount()` from the account logic as it will now be triggered in the liquidator contract.
-Added `checkAndStartLiquidation()` in `AccountV1.sol`. This function will be called by the liquidator contract to initiate a liquidation. The function should revert if an Account is not liquidatable.

### CODE REVIEW

File: src/AccountV1.sol [hunk 1]
```diff
@@ -443,50 +452,46 @@ contract AccountV1 is AccountStorageV1, IAccount {
...
+     * @return assetIds Array of the IDs of the assets in Account.
+     * @return assetAmounts Array with the amounts of the assets in Account.
+     * @return creditor_ The trusted creditor, address 0 if no active trusted creditor.
+     * @return totalOpenDebt The total open Debt against the Account.
+     * @return assetAndRiskValues Array of asset values and corresponding collateral factors.
      */
-    function liquidateAccount(uint256 openDebt)
+    function checkAndStartLiquidation()
         external
-        nonReentrant
```
[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/AccountV1.sol:456]: Why is the reentrance guard deleted?
Liquidations must be blocked during the doAccountManagementActions

File: test/Base.t.sol [hunk 2]
```diff
@@ -124,9 +124,10 @@ abstract contract Base_Test is Test, Events, Errors {
         trustedCreditor.setLiquidator(Constants.initLiquidator);
 
         // Deploy an initial Account with all inputs to zero
-        vm.prank(users.accountOwner);
+        vm.startPrank(users.accountOwner);
```
Comment (Thomas-Smets) [inline test/Base.t.sol:127]: Why this change?

File: test/fuzz/accounts/AccountV1/CheckAndStartLiquidation.fuzz.t.sol [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        // Mint and approve Stable1 tokens
+        vm.startPrank(users.tokenCreatorAddress);
+        mockERC20.stable1.mint(users.accountOwner, depositAmountStable1);
+        vm.startPrank(users.accountOwner);
+        mockERC20.stable1.approve(address(accountExtension2), type(uint256).max);
+
+        // Deposit stable1 token in account
+        accountExtension2.deposit(assetAddresses, assetIds, assetAmounts);
+
+        // Then : Account should not be liquidatable as openDebt > 0 and liquidationValue > usedMargin
```
Comment (Thomas-Smets) [inline test/fuzz/accounts/AccountV1/CheckAndStartLiquidation.fuzz.t.sol:97]: If you use Given-When-Then for the unit tests, be consistent and use all three.

File: test/fuzz/accounts/AccountV1/LiquidateAccount.fuzz.t.sol [hunk 4]
```diff
@@ -1,103 +0,0 @@
...
-        AccountV1_Fuzz_Test.setUp();
-
-        // Given: Trusted Creditor is set.
-        openMarginAccount();
-    }
-
-    /*//////////////////////////////////////////////////////////////
-                              TESTS
-    //////////////////////////////////////////////////////////////*/
-    function testFuzz_Revert_liquidateAccount_Reentered(uint128 debt) public {
```
Comment (Thomas-Smets) [inline test/fuzz/accounts/AccountV1/LiquidateAccount.fuzz.t.sol:27]: We still need this

File: src/AccountV1.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (acollette) [reply src/AccountV1.sol:456]: That was for the  liquidateAccount function that is now triggered in the Liquidator contract.

[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/AccountV1.sol:456]: Doesn't matter where it is triggered, the Reentrance vulnerability is in the Account, so the reentrancy guard must be in the Account

File: test/Base.t.sol (hunk 2 above)
Comment (acollette) [reply test/Base.t.sol:124]: I saw it's good practice to use` changePrank() `which we can now do by using `startPrank()` on a current prank. (as `changePrank()` got deprecated). And thus no need always to call `stopPrank()`, you can just prank a new address by using `startPrank()` again.

File: test/fuzz/accounts/AccountV1/LiquidateAccount.fuzz.t.sol (hunk 4 above)
Comment (acollette) [reply test/fuzz/accounts/AccountV1/LiquidateAccount.fuzz.t.sol]: reintroduced liquidateAccount() for tests to pass in lending repo. To remove later.
